File: crafting_unit_tests/code/java/first_sample_1.java
@Test
public void should_return_2018_template() {
   if (LocalDateTime.now().getYear() == 2018){
       assertThat(getVersion()).isEqualTo("2018.1.7");
   }
   else{
       assertThat(getVersion()).isEqualTo("1.7");
   }
}

// .....................................

public String getVersion() {
   if (LocalDateTime.now().getYear() >= 2018){
       return LocalDateTime.now().getYear() + ".1.7";
   }
   else{
       return "1.7";
   }
}

File: crafting_unit_tests/code/python/first_sample_1.py

@Test
def should_return_2018_template(self):
   if LocalDateTime.now().getYear() == 2018:
       self.assertEqual("2018.1.7", getVersion())
   else:
       self.assertEqual("1.7", getVersion())

# .....................................

def getVersion():
   if now().year >= 2018:
       return now().year + ".1.7";
   else:
       return "1.7";
